langchain/vectorstores/supabase.py

```diff
@@ -335,3 +335,95 @@ def max_marginal_relevance_search(
             embedding[0], k, fetch_k, lambda_mult=lambda_mult
         )
         return docs
+
+    def add_documents_by_id(
+        self, documents: List[Document], ids: List[str]
+    ) -> List[str]:
+        """Add the documents to vectorstore by ID.
+
+        Args:
+            documents: A sequence of documents to add.
+            ids: A sequence of ids to use for the documents.
+
+        Returns:
+            A list of ids of the added documents. Auto-assigned if not provided.
+        """
+        texts = [document.page_content for document in documents]
+        metadatas = [document.metadata for document in documents]
+        embeddings = self._embedding.embed_documents(texts)
+        rows: List[dict[str, Any]] = [
+            {
+                "content": text,
+                "embedding": embedding,
+                "id": id,
+                "metadata": metadata,
+            }
+            for text, metadata, id, embedding in zip(texts, metadatas, ids, embeddings)
+        ]
+
+        # Handle each insert individually to avoid conflicting IDs
+        for row in rows:
```

Comment:
This is unfortunate. May slow adds considerably. Need to see if there is a better way.